### ISSUE TITLE
Natshare throwing null reference exceptions in Editor

### ISSUE THREAD

**Issue by Klipi** [natsuite/NatShare-API#77]
I recently updated to the latest NatShare v.1.2.1. The editor fallback throws a null reference exception whenever `Dispose` is called on the `SharePayload` object.

Here's a minimal example:

```
public void OnShareClicked()
{
    using (var payload = new SharePayload())
    {
        payload.AddText("Exception!");
    }
}
```

The cause seems to be that the native payload is set to null, after which `Dispose` is called on a null object:

```
// In the constructor of SharePayload
default:
    Debug.LogError("NatShare Error: SharePayload is not supported on this platform");
    this.payload = null; // Self-destruct >:D
    break;

/// <summary>
/// Dispose the payload
/// </summary>
[Doc(@"Dispose")]
public void Dispose () => payload.Dispose();
```

I'd suggest doing a minimal stub of the `ISharePayload` interface that gracefully fails to share in the Editor. Do you accept pull requests?

Edit:

The full exception:

```
NullReferenceException: Object reference not set to an instance of an object
NatShare.SharePayload.Dispose () (at Assets/Third Party/NatShare/Plugins/Managed/SharePayload.cs:50)
MyBehaviour.OnShareClicked () (at Assets/Scripts/MyBehaviour.cs:55)
UnityEngine.Events.InvokableCall.Invoke () (at /Users/builduser/buildslave/unity/build/Runtime/Export/UnityEvent/UnityEvent.cs:166)
UnityEngine.Events.UnityEvent.Invoke () (at /Users/builduser/buildslave/unity/build/Runtime/Export/UnityEvent/UnityEvent/UnityEvent_0.cs:58)
UnityEngine.UI.Button.Press () (at /Applications/Unity/Hub/Editor/2019.2.12f1/Unity.app/Contents/Resources/PackageManager/BuiltInPackages/com.unity.ugui/Runtime/UI/Core/Button.cs:68)
UnityEngine.UI.Button.OnPointerClick (UnityEngine.EventSystems.PointerEventData eventData) (at /Applications/Unity/Hub/Editor/2019.2.12f1/Unity.app/Contents/Resources/PackageManager/BuiltInPackages/com.unity.ugui/Runtime/UI/Core/Button.cs:110)
UnityEngine.EventSystems.ExecuteEvents.Execute (UnityEngine.EventSystems.IPointerClickHandler handler, UnityEngine.EventSystems.BaseEventData eventData) (at /Applications/Unity/Hub/Editor/2019.2.12f1/Unity.app/Contents/Resources/PackageManager/BuiltInPackages/com.unity.ugui/Runtime/EventSystem/ExecuteEvents.cs:50)
UnityEngine.EventSystems.ExecuteEvents.Execute[T] (UnityEngine.GameObject target, UnityEngine.EventSystems.BaseEventData eventData, UnityEngine.EventSystems.ExecuteEvents+EventFunction`1[T1] functor) (at /Applications/Unity/Hub/Editor/2019.2.12f1/Unity.app/Contents/Resources/PackageManager/BuiltInPackages/com.unity.ugui/Runtime/EventSystem/ExecuteEvents.cs:261)
UnityEngine.EventSystems.EventSystem:Update() (at /Applications/Unity/Hub/Editor/2019.2.12f1/Unity.app/Contents/Resources/PackageManager/BuiltInPackages/com.unity.ugui/Runtime/EventSystem/EventSystem.cs:377)
```

**Comment by olokobayusuf** [comment]
Fixed in #71 (293370eda369a076a0008e7422a3294d13cf3f06).

**Comment by olokobayusuf** [comment]
And yes, PR's are always welcome!
